### ISSUE TITLE
[java] New rule: UnnecessaryVarargsArrayCreation

### ISSUE THREAD

**Issue by oowekyala** [pmd/pmd#3216]
**Proposed Rule Name:** UnnecessaryVarargsArrayCreation

**Proposed Category:** ~Code Style~ Best Practices

**Description:** This new rule would flag instances of code that explicitly creates an array when calling a varargs method. This is more heavy to read and could be simplified.

I think this new rule may only be written in pmd 7, because type resolution in pmd 6 does not expose information about the called method.

**Code Sample:**

```java
List<String> strings = Arrays.asList(new String[] { "a", "b", }); // flagged by the rule
strings = Arrays.asList("a", "b"); // ok
```

```java
Arrays.asList(someVariable) // ok
```

**Possible Properties:**

*   *Should this rule be customizable via properties?* No need
